src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -462,6 +462,8 @@ private MsalAccessTokenCacheItem FilterByExpiry(MsalAccessTokenCacheItem msalAcc
                         "Access token has expired or about to expire. " +
                         GetAccessTokenExpireLogMessageContent(msalAccessTokenCacheItem));
                 }
+
+                requestParams.RequestContext.ApiEvent.CacheRefresh = "1";
```

Comment:
I think the logic for values "0" and "1" should be done here: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/ccbf33200ba4ddb606f5a3e13b05c9c8e43e8335/src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs#L328

Reply:
Smth like

```csharp

  // no match
            if (finalList.Count == 0)
            {
                logger.Verbose("No tokens found for matching authority, client_id, user and scopes.");
                API_EVENT = "0"
                return null;
            }
  
            msalAccessTokenCacheItem = FilterByExpiry(msalAccessTokenCacheItem, requestParams);
            if (at==null) { API_EVENT = "1" }
```

Reply:
Telemetry (and logging) are cross cutting concerns, meaning that it's ok for all parts of the product to contribute to telemetry. This violates Single Responsability Principle, but it's acceptable. Still,  let's try to concentrate as much of this logic to one place rather than spreading it around.